test/e2e/create_test.go

```diff
@@ -602,3 +602,38 @@ func (tc *testContext) scaleMachineApprover(replicas int) error {
 	return tc.scaleDeployment("openshift-cluster-machine-approver", "machine-approver", "app=machine-approver",
 		&replicaCount)
 }
+
+// createPullSecret creates a pull secret in the operator test namespace
+func (tc *testContext) createPullSecret() error {
+	pullSecret, err := tc.client.K8s.CoreV1().Secrets("openshift-config").Get(context.TODO(),
+		"pull-secret", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to retrieve pull secret: %w", err)
+	}
+
+	_, err = tc.client.K8s.CoreV1().Secrets(tc.workloadNamespace).Get(context.TODO(),
+		"pull-secret", metav1.GetOptions{})
+	if err == nil {
+		return nil
+	}
+
+	if !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to retrieve pull secret: %w", err)
+	}
```

Comment:
This should be at the beginning of the function, as there is no point in getting the pull-secret if this already exists